### PR TITLE
FPDF.image() now handles keep_aspect_ratio in combination with an enum value provided to argument x

### PR DESCRIPTION
**Pull request by Lucas-C**
**Checklist**:

- [x] The GitHub pipeline is OK (green), meaning that both `pylint` (static code analyzer) and `black` (code formatter) are happy with the changes of this PR.

- [x] A unit test is covering the code added / modified by this PR

- [x] This PR is ready to be merged

- [x] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder

- [x] A mention of the change is present in `CHANGELOG.md`

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/py-pdf/fpdf2/blob/master/LICENSE).

### CODE REVIEW

**Comment by Lucas-C** [comment]
If you want to review this @gmischler & @andersonhc, I'll be happy to have your feedbacks 🙂

Else I'll merge this in a week or so.

File: docs/Text.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ For all text insertion methods, the relevant font related properties (eg. font/s
 * [`.set_draw_color()`](fpdf/fpdf.html#fpdf.fpdf.FPDF.set_draw_color) - for cell borders
 * [`.set_fill_color()`](fpdf/fpdf.html#fpdf.fpdf.FPDF.set_fill_color) - for the background
 
-All those methods accept either a single greyscale value, 3 valeurs as RGB components, a single `#abc` or `#abcdef` hexadecimal color string, or an instance of [`fpdf.drawing.DeviceCMYK`](https://py-pdf.github.io/fpdf2/fpdf/drawing.html#fpdf.drawing.DeviceCMYK), [`fpdf.drawing.DeviceRGB`](https://py-pdf.github.io/fpdf2/fpdf/drawing.html#fpdf.drawing.DeviceRGB) or [`fpdf.drawing.DeviceGray`](https://py-pdf.github.io/fpdf2/fpdf/drawing.html#fpdf.drawing.DeviceGray).
+All three `set_*_colors()` methods accept either a single greyscale value, 3 valeurs as RGB components, a single `#abc` or `#abcdef` hexadecimal color string, or an instance of [`fpdf.drawing.DeviceCMYK`](https://py-pdf.github.io/fpdf2/fpdf/drawing.html#fpdf.drawing.DeviceCMYK), [`fpdf.drawing.DeviceRGB`](https://py-pdf.github.io/fpdf2/fpdf/drawing.html#fpdf.drawing.DeviceRGB) or [`fpdf.drawing.DeviceGray`](https://py-pdf.github.io/fpdf2/fpdf/drawing.html#fpdf.drawing.DeviceGray).
```
Comment (andersonhc) [inline docs/Text.md:49]: Small typo. Replace *3 valeurs* by *3 values*

File: fpdf/fpdf.py [hunk 2]
```diff
@@ -4185,8 +4189,21 @@ def _raster_image(
         self.images_used_per_page_number[self.page].add(info["i"])
         return RasterImageInfo(**info, rendered_width=w, rendered_height=h)
 
+    def x_by_align(self, x, w, h, img_info, keep_aspect_ratio):
```
Comment (gmischler) [inline fpdf/fpdf.py:4192]: Why did you move this out of `ImageInfo`?
(now that it actually uses another method from there...)

fpdf.py is already way too crowded. Specialized functionality should be delegated elsewhere whenever possible.

**Review by gmischler — APPROVED** [review]
Only minor nitpicks this time.

Otherwise those are very helpful fixes and additions!

File: fpdf/fpdf.py (hunk 2 above)
Comment (Lucas-C) [reply fpdf/fpdf.py:4192]: > fpdf.py is already way too crowded. Specialized functionality should be delegated elsewhere whenever possible.

I agree.

However, the code of this method depends on one line on `img_info`, and instance of `ImageInfo`,
and on **three lines** on an instance of `FPDF`! (to access `.w`, `.r_margin`, `.l_margin`)

Given that this method is only invoked in two places, inside `FPDF._*_image()` methods,
it really did not make much sense to me to keep inside `ImageInfo`...

File: docs/Text.md (hunk 1 above)
Comment (Lucas-C) [reply docs/Text.md]: Thank you, fixed!

File: fpdf/fpdf.py (hunk 2 above)
Comment (Lucas-C) [reply fpdf/fpdf.py:4192]: I choosed to merge this PR as it is,
but I agree that we could try to move this method elswhere in another PR.

I'm not sure that `ImageInfo` is the best place though...

We could also have `FPDF` implement a `PageDimension` interface,
that provides `.w`, `.h`, `.epw`, `.eph`, `.t_margin` `.b_margin`, `.l_margin`, `.r_margin`...
This way `FPDF` instances could be provided to methods typed as `PageDimension`,
to reduce the coupling and make it explicit that only the page dimensions contained in `FPDF` instances is used.
